idx/memory/memory.go

```diff
@@ -453,6 +460,11 @@ func (m *UnpartitionedMemoryIdx) add(def *schema.MetricDefinition) idx.Archive {
 		return *archive
 	}
 
+	// invalidate any findCache items that match this series name
+	defer func() {
```

Comment:
it should never be up to a call site to document what the function being called does.
so, remove this comment, and to `FindCache.InvalidateFor` add a comment explaining what exactly it does (which can be in more detail, seems like it clears every single subtree as well)

Reply:
updated